test/FsAutoComplete.Tests.Lsp/CodeFixTests.fs

```diff
@@ -1263,6 +1263,987 @@ let private generateUnionCasesTests state =
+              { 
+                new IDisposable
+                with
+                  member _.Dispose() = failwith "-"
+              }
+              """
+          testCaseAsync "with 3 lines below with comments" <|
+            CodeFix.check server
+              """
+              open System
+              { 
+                new IDis$0posable
+                // some
+                // comment
+                with
+              }
+              """
+              validateDiags
+              selectCodeFixWithoutTypeAnnotation
+              """
+              open System
+              { 
+                new IDisposable
+                // some
+                // comment
+                with
+                  member _.Dispose() = failwith "-"
+              }
+              """
+          testCaseAsync "new on prev line" <|
+            CodeFix.check server
+              """
+              open System
+              { 
+                new 
+                      IDis$0posable with
+              }
+              """
+              validateDiags
+              selectCodeFixWithoutTypeAnnotation
+              """
+              open System
+              { 
+                new 
+                      IDisposable with
+                  member _.Dispose() = failwith "-"
+              }
+              """
+          testCaseAsync "new and with on extra lines" <|
+            CodeFix.check server
+              """
+              open System
+              { 
+                new 
+                  IDis$0posable 
+                with
+              }
+              """
+              validateDiags
+              selectCodeFixWithoutTypeAnnotation
+              """
+              open System
+              { 
+                new 
+                  IDisposable 
+                with
+                  member _.Dispose() = failwith "-"
+              }
+              """
+        ]
+      ]
+    ])
+    let config = { 
+      defaultConfigDto with 
+        IndentationSize = Some 6
+        InterfaceStubGeneration = Some true 
+        InterfaceStubGenerationObjectIdentifier = Some "this"
+        InterfaceStubGenerationMethodBody = Some "raise (System.NotImplementedException())"
+    }
+    serverTestList "with 6 indentation" state config None (fun server -> [
+      let testBoth = testBoth server
+      testBoth "uses indentation, object identifier & method body from config"
+          """
+          type X() =
+                interface System.$0IDisposable with
+          """
+          """
+          type X() =
+                interface System.IDisposable with
+                      member this.Dispose(): unit = 
+                            raise (System.NotImplementedException())
+          """
+          """
+          type X() =
+                interface System.IDisposable with
+                      member this.Dispose() = raise (System.NotImplementedException())
+          """
+      ()
```

Comment:
_excellent_ test coverage for these changes! I'm so happy that the work you did on the codefix helpers makes it easy to add new tests.